### PR TITLE
fix: cloud events - convert kafka prefix

### PR DESCRIPTION
**Pull request by octonato**
Convert events are using `ce_` prefix, we convert to `ce-`

### CODE REVIEW

**Comment by franciscolopezsancho** [comment]
I'm curious. Why so?

**Comment by franciscolopezsancho** [comment]
Right. Now I know (after the standup)

File: sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/MetadataImpl.scala [hunk 1]
```diff
@@ -291,4 +302,17 @@ object MetadataImpl {
...
+  def of(entries: Seq[MetadataEntry]): MetadataImpl = {
+    val transformedEntries =
+      entries.map { entry =>
+        // is incoming ce key in one of the alternative formats?
+        // if so, convert key to our internal default key format
+        alternativeKeyFormats.get(entry.key) match {
+          case Some(defaultKey) => MetadataEntry(defaultKey, entry.value)
+          case _                => entry
+        }
+      }
```
Comment (octonato) [inline sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/MetadataImpl.scala:314]: I changes this implementation. 
Before, I was replacing any key starting with `ce_` with a key starting with `ce-`. 

This new implementation creates a Map for all know keys,like `ce_id` maps to `ce-id`. If we receive a `ce_id`, we transform it. 

If we receive keys starting with `ce_`, but not in the list of known keys, we don't touch them.

File: sdk/java-sdk-protobuf/src/test/scala/kalix/javasdk/impl/MetadataImplSpec.scala [hunk 2]
```diff
@@ -159,10 +159,20 @@ class MetadataImplSpec extends AnyWordSpec with Matchers with OptionValues {
       val mdRedirect = md.withStatusCode(Redirect.MOVED_PERMANENTLY)
       mdRedirect.get("_kalix-http-code").toScala.value shouldBe "301"
     }
+
+    "support creationg with CloudEvents prefixed with ce_" in {
```
Comment (franciscolopezsancho) [inline sdk/java-sdk-protobuf/src/test/scala/kalix/javasdk/impl/MetadataImplSpec.scala:163]: Shall we add a test with metadata prefixed with `ce-`?

Comment (franciscolopezsancho) [reply sdk/java-sdk-protobuf/src/test/scala/kalix/javasdk/impl/MetadataImplSpec.scala:163]: oh, I see. It was already working for `ce-`.

**Review by franciscolopezsancho — APPROVED** [review]
LGTM

**Pull request merged**